api/revisions/serializers.py

```diff
@@ -0,0 +1,154 @@
+    reviews_state = ser.ChoiceField(choices=['revision_in_progress', 'revision_pending_admin_approval', 'revision_pending_moderation', 'approved'], required=False)
+    is_pending_current_user_approval = ser.SerializerMethodField()
+
+    links = LinksField(
+        {
+            'self': 'get_absolute_url',
+        },
+    )
+
+    registration = RelationshipField(
+        related_view='registrations:registration-detail',
+        related_view_kwargs={'node_id': '<parent._id>'},
+        required=False,
+    )
+
+    registration_schema = RelationshipField(
+        related_view='schemas:registration-schema-detail',
+        related_view_kwargs={'schema_id': '<parent.registered_schema_id>'},
+        read_only=True,
+        required=False,
+    )
+
+    initiated_by = RelationshipField(
+        related_view='users:user-detail',
+        related_view_kwargs={'user_id': '<initiator._id>'},
+        read_only=True,
+        required=False,
+
+    )
+
+    class Meta:
+        type_ = 'revisions'
+
+    def get_absolute_url(self, obj):
+        return absolute_reverse(
+            'revisions:schema-responses-detail',
+            kwargs={
+                'version': self.context['request'].parser_context['kwargs']['version'],
+                'revision_id': obj._id,
+            },
+        )
+
+    def get_is_pending_current_user_approval(self, obj):
+        # TBD
+        return False
+
+
+class SchemaResponsesListSerializer(SchemaResponsesSerializer):
+
+    def create(self, validated_data):
+        registration = Registration.load(validated_data.pop('_id'))
+        if registration.registered_schema.first():
+            schema_response = SchemaResponses.objects.create(
+                **validated_data,
+                object_id=registration.id,
+                content_type=ContentType.objects.get_for_model(registration),
+                initiator=self.context['request'].user
+            )
+        else:
+            raise NotImplementedError()
+
+        return schema_response
+
+
+class SchemaResponsesDetailSerializer(SchemaResponsesSerializer):
+
+    versions = RelationshipField(
+        related_view='registrations:schema-responses-list',
+        related_view_kwargs={'node_id': '<node._id>'},
+    )
+    writeable_method_fields = frozenset([
+        'revision_response',
+    ])
+    revision_response = ser.SerializerMethodField()
+    revised_responses = ser.SerializerMethodField()
+
+    def get_revision_response(self, obj):
+        data = []
+        for response_block in obj.response_blocks.all():
+            data.append({response_block.schema_key: response_block.response})
+        return data
+
+    def get_revised_responses(self, obj):
+        previous_version = obj.previous_version
+
+        if not previous_version:
+            return []
+        else:
+            return obj.response_blocks.values_list('schema_key', flat=True)
+
+    def update(self, revision, validated_data):
```

Comment:
Can you leave this unimplemented for now? I have strong enough thoughts that I'd like this and the nuances of the create functionality can be their own PR that we can pair on.